### PR TITLE
Devel

### PR DESCRIPTION
**Pull request by heidydallard**

### CODE REVIEW

File: include/gepetto/gui/mainwindow.hh [hunk 1]
```diff
@@ -17,6 +17,7 @@
 #include <gepetto/gui/dialog/pluginmanagerdialog.hh>
 
 #include <gepetto/gui/settings.hh>
+#include <gepetto/gui/shortcut-factory.hh>
```
Comment (jmirabel) [inline include/gepetto/gui/mainwindow.hh]: To reduce compilation time, please:
- predefine `ShortcutFactory` in `gepetto/gui/fwd.hh`
- move `#include <gepetto/gui/shortcut-factory.hh>` to `mainwindow.cc`

File: include/gepetto/gui/mainwindow.hh [hunk 2]
```diff
@@ -155,7 +156,7 @@ signals:
 
         void connectSlot(const char *slot, const char *signal, QObject* obj);
 
-        /// \}
```
Comment (jmirabel) [inline include/gepetto/gui/mainwindow.hh]: Do not delete this, you would mess up the documentation.

File: include/gepetto/gui/mainwindow.hh [hunk 3]
```diff
@@ -155,7 +156,7 @@ signals:
...
+      void registerShortcut(QString widgetName, QString actionName, QAction* action);
```
Comment (jmirabel) [inline include/gepetto/gui/mainwindow.hh]: See line note in https://github.com/jmirabel/hpp-plot/pull/5#discussion_r71306052

File: include/gepetto/gui/mainwindow.hh [hunk 4]
```diff
@@ -157,6 +157,9 @@ signals:
 
         /// \}
 
+      void registerShortcut(QString widgetName, QString actionName, QAction* action);
+      void registerShortcut(QString widgetName, QAction* action);
```
Comment (jmirabel) [inline include/gepetto/gui/mainwindow.hh]: To keep the documentation consistent, could you put that in the section `available_in_python` ?

File: include/gepetto/gui/shortcut-factory.hh [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      ~ShortcutFactory();
+
+    public slots:
+      /// Add a binding to a given widget.
+      /// If widgetName doesn't exists, then it will be add.
+      /// If actionName exists, do nothing
+      /// \param widgetName name of the corresponding widget
+      /// \param actionName name of the action
+      /// \param action instance of the action
+      void addBinding(QString widgetName, QString actionName, QAction* action);
```
Comment (jmirabel) [inline include/gepetto/gui/shortcut-factory.hh]: Is it used as a slot ? What is the reason for declaring it as a slot ?

Comment (heidydallard) [inline include/gepetto/gui/shortcut-factory.hh]: At the beginning it was supposed to be a slot but then I changed my mind and forgot to change it.
I changed it.

**Comment by jmirabel** [comment]
Thanks !

**Pull request merged**